File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/match/MatchDao.java
package com.legue.axel.lolsummonertool.database.dao.match;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.legue.axel.lolsummonertool.database.model.match.Match;

import java.util.List;

@Dao
public interface MatchDao {

    @Query("SELECT * FROM matches ORDER BY gameId")
    LiveData<List<Match>> getMatches();

    @Query("SELECT * FROM matches WHERE gameId = :gameId")
    LiveData<Match> getMatche(long gameId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllMatches(List<Match> matches);

    @Query("DELETE FROM matches")
    void deleteAll();

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateMatch(Match match);

}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/match/MatchDao.kt
package com.legue.axel.lolsummonertool.database.dao.match

import androidx.lifecycle.LiveData
import androidx.room.*
import com.legue.axel.lolsummonertool.database.model.match.Match

@Dao
interface MatchDao {

    @Query("SELECT * FROM matches ORDER BY gameId")
    fun getMatches(): LiveData<List<Match>>

    @Query("SELECT * FROM matches WHERE gameId = :gameId")
    fun getMatche(gameId: Long): LiveData<Match>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllMatches(matches: List<Match>)

    @Query("DELETE FROM matches")
    fun deleteAll()

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateMatch(match: Match)

}
